plugins/mumble_plugin_linux.h

```diff
@@ -151,11 +148,45 @@ static inline PTR_TYPE_CONCRETE getModuleAddr(pid_t pid, const wchar_t *modname)
 	return 0;
 }
 
-static inline PTR_TYPE_CONCRETE getModuleAddr(const wchar_t *modname) {
+static inline procptr_t getModuleAddr(const wchar_t *modname) {
 	return getModuleAddr(pPid, modname);
 }
 
-static inline bool peekProc(PTR_TYPE base, void *dest, size_t len) {
+static inline int checkProcessIs64Bit(const pid_t pid)
+{
+	// We can know the process architecture by looking at its ELF header.
+	// The fifth byte is 1 in case the process is 32-bit or 2 in case it's 64-bit.
+	// The first 4 bytes constitute the magical number in ASCII: 0x7F 45 4c 46.
+
+	std::stringstream ss;
+	ss << std::string("/proc/");
+	ss << static_cast<unsigned long>(pid);
+	ss << std::string("/exe");
+
+	std::ifstream ifs;
+	ifs.open(ss.str(), std::ifstream::binary);
+
+	std::string elf;
+
+	for (int i = 0; i < 5; i++) {
+		char buf;
+		ifs.read(&buf, sizeof(buf));
+		size_t nread = ifs.gcount();
+		if (nread > 0) {
+			elf.append(&buf, nread);
+		}
+	}
+
+	ifs.close();
+
+	if (!(elf.at(0) == 0x7f && elf.at(1) == 'E' && elf.at(2) == 'L' && elf.at(3) == 'F')) {
+		return -1;
```

Comment:
Does -1 indicate failure to identify then? Please document the function.

Reply:
Right.